src/S3/S3MultiRegionClient.php

```diff
@@ -269,6 +286,11 @@ public function getObjectUrl($bucket, $key)
         return $regionalClient->getObjectUrl($bucket, $key);
     }
 
+    public function getDetermineBucketRegionGenerator($bucketName)
```

Comment:
Why use this method instead of just directly calling `yield $this->determineBucketRegionAsync($bucketName);`? If the intention is just to avoid repeating `yield`, I think this should be private